selfdrive/ui/qt/offroad/wifiManager.cc

```diff
@@ -277,7 +276,19 @@ void WifiManager::stateChange(unsigned int new_state, unsigned int previous_stat
 
 // https://developer.gnome.org/NetworkManager/stable/gdbus-org.freedesktop.NetworkManager.Device.Wireless.html
 void WifiManager::propertyChange(const QString &interface, const QVariantMap &props, const QStringList &invalidated_props) {
-  if (interface == NM_DBUS_INTERFACE_DEVICE_WIRELESS && props.contains("LastScan")) {
+  if (props.contains("ActiveConnection")) {
+    QString path = props.value("ActiveConnection").value<QDBusObjectPath>().path();
+    if (emptyPath(path)) {
+      setCurrentSsid("");
+    } else {
+      auto active_conn_object = call<QDBusObjectPath>(path, NM_DBUS_INTERFACE_PROPERTIES, "Get", NM_DBUS_INTERFACE_ACTIVE_CONNECTION, "SpecificObject");
+      auto active_conn_state = call<uint>(path, NM_DBUS_INTERFACE_PROPERTIES, "Get", NM_DBUS_INTERFACE_ACTIVE_CONNECTION, "State");
+      if (active_conn_state == NM_ACTIVE_CONNECTION_STATE_ACTIVATING && !emptyPath(active_conn_object.path())) {
+        setCurrentSsid(get_property(active_conn_object.path(), "Ssid"));
+      }
+    }
+
+  } else if (interface == NM_DBUS_INTERFACE_DEVICE_WIRELESS && props.contains("LastScan")) {
```

Comment:
@deanlee it would be nice to unify the "ACTIVATED" part of `stateChange` into this, know of any way to delete that part?

Reply:
I only see these props on activate, doesn't look like it:

```
selfdrive/ui/qt/offroad/wifiManager.cc: props: QMap(("Dhcp4Config", QVariant(QDBusObjectPath, )))

selfdrive/ui/qt/offroad/wifiManager.cc: props: QMap(("State", QVariant(uint, 80)))
selfdrive/ui/qt/offroad/wifiManager.cc: props: QMap(("StateReason", QVariant(QDBusArgument, )))
selfdrive/ui/qt/offroad/wifiManager.cc: pror.cc: props: QMap(("Dhcp4Config", QVariant(QDBusObjectPath, )))
selfdrive/ui/qt/offroad/wifiManager.cc: props: QMap(("Ip6Config", QVariant(QDBusObjectPath, )))
selfdrive/ui/qt/offroad/wifiManager.cc: props: QMap(("Dhcp6Config", QVariant(QDBusObjectPath, )))
selfdrive/ui/qt/offroad/wifiManager.cc: props: QM("State", QVariant(uint, 100)))
selfdrive/ui/qt/offroad/wifiManager.cc: props: QMap(("StateReason", QVariant(QDBusArgument, )))
selfdrive/ui/qt/offroad/wifiManager.cc: props: QMap(("Bitrate", QVariant(uint, 65100)))
selfdrive/ui/qt/offroad/wifiManager.cc: props: QMap(("Metered", QVariant(uint, 4)))
selfdrive/ui/qt/offroad/wifiManager.cc: props: QMap(("Ip4Connectivity", QVariant(uint, 4)))
selfdrive/ui/qt/offroad/wifiManager.cc: props: QMap(("Ip6Connectivity", QVariant(uint, 4)))
```